bitsv/wallet.py

```diff
@@ -14,26 +14,41 @@
     )
 from bitsv import op_return
 
+# Instantiate Main, Test and STN NetworkAPI instances for use by all PrivateKey[Test]'s
+network_api_main = NetworkAPI('main')
+network_api_test = NetworkAPI('test')
+network_api_stn = NetworkAPI('stn')
+
+
+def wif_to_key(wif, testnet='test'):
+    """This function can read the 'version' byte of a wif and instatiate the appropriate PrivateKey object.
+
+    Because the version byte is the same for testnet and scaling-testnet, the testnet='test' parameter has been added
+    to specify which test network to use. i.e. 'test' or 'stn'.
+
+    By default this function will work as it always used to and select either 'main' or 'test' networks.
+
+    :param wif: A private key serialized to the Wallet Import Format. If the
+                argument is not supplied, a new private key will be created.
+                The WIF compression flag will be adhered to, but the version
+                byte is disregarded. Compression will be used by all new keys.
+    :type wif: ``str``
+    :param testnet: 'test' or 'stn'
```

Comment:
I really don't like this. It's very unintuitive and hacky.

I think what I would do is have `wif_to_key(wif, chain='main' (or maybe None for auto))`

So if version is None, auto detect as usual. If version is set, make sure the wif matches the expected format. So if someone does wif='main' and passes a testnet or stn key, we get ValueError.

Reply:
Okay. Thanks. I think I can make this a lot better on that front then. I have family staying until Saturday but I'll work on it Saturday.

Reply:
Yes, I get that. Auto will only give you main or testnet, not stn. If you want stn, you have to ask for it (and then it only works when it autodetects a testnet-style WIF).

Take your time, I hope you enjoy your time with your family.

Reply:
Okay I've had a crack at implementing these suggested changes. I think it is a lot better than it was before.
- Let me know what you think.
- Make sure you look at the final pooled changes because I did something a bit dumb with my first commit... but then I fixed it when I fixed the tests...

Reply:
Nice work! I would still add tests but the new code is really lengthy. I thought it didn't need to be nearly that long for some reason. Let me try to write up the logic and see if it works in something more concise.

Reply:
Haven't tested this, but please let me know what you think.

```
def wif_to_key(wif, network=None):
    """This function can read the 'prefix' byte of a wif and instatiate the appropriate PrivateKey object.
    see: https://en.bitcoin.it/wiki/List_of_address_prefixes
    The prefix byte is the same for testnet and scaling-testnet.
    So to use scaling-testnet you must give as a parameter: network='stn'
    :param wif: A private key serialized to the Wallet Import Format.
    :type wif: ``str``
    :param network: 'main', 'test' or 'stn'
    :type network: ``str``
    """
    private_key_bytes, compressed, prefix = wif_to_bytes(wif)

    wif_network_mismatch = "WIF prefix: '{}' does not match network: '{}'".format(prefix, network)

    if network == 'main':
        if prefix != 'main':
            raise ValueError(wif_network_mismatch)
    elif network in ['test', 'stn']:
        if prefix != 'test':
            raise ValueError(wif_network_mismatch)
    elif network is None:
        if prefix == 'main':
            network = 'main'
        elif prefix == 'test':
            network = 'test'
        else:
            raise Exception('bitsv issue, please open a bug report!')
    else:
        raise ValueError('network must be one of: main, test, stn')

    if compressed:
        return PrivateKey.from_bytes(private_key_bytes, network=network)
    else:
        return PrivateKey(wif, network=network)
```

Reply:
Read this on my phone but yes that looks much cleaner. Thanks. 😉

Reply:
Okay @teran-mckinney I have implemented this change now. Passes tests.

Reply:
Awesome, no problem!